dali/operators/reader/loader/tar_utils.h

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_TAR_UTILS_H_
+#define DALI_OPERATORS_READER_LOADER_TAR_UTILS_H_
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <vector>
+#include "dali/core/common.h"
+#include "dali/util/file.h"
+
+namespace dali {
+namespace detail {
+class TarArchive {
+ public:
+  TarArchive() = default;
+  explicit TarArchive(std::unique_ptr<FileStream>&& stream);
```

Comment:
```suggestion
  explicit TarArchive(std::unique_ptr<FileStream> stream);
```
unique_ptr is move-only, so there can't be a copy and the caller needs to pass an r-value anyway. The reasons to use `&&` could be twofold, but I think neither applies:
1. Exception handling - if the function fails to complete (prior to moving out the contents of the pointer), the caller retains the original object and may handle the exception and enjoy the ownership of the object.
2. Conditional move - passing an object by `&&` enables the function to move out the contents from the original object, but does not mandate it - under some circumstances (defined by the particular use case) , the function may decide not to move out the object and the caller will retain ownership. If we're not interested in this scenario - or indeed, we want to guarantee that the object is moved out and either moved out by the callee or destroyed - then we should pass a move-only object by value, because this guarantees that the object will be first moved to the argument.

Reply:
You wouldn't.